### DISCUSSION TITLE
Connect with the device name

### DISCUSSION THREAD

**Discussion by enzo69006** [innoveit/react-native-ble-manager#946] [Q&A] [answered]
Hello, it is possible to use BleManager.connect('nameOfPeripherals') to enable the connection ?
Because since 2 days, i can only connect with the mac address of my peripherals.

**Comment by marcosinigaglia** [comment, marked as answer]
No, is not possible, the name is not unique.

**Reply by enzo69006** [reply]
No alternative ?

**Reply by marcosinigaglia** [reply]
No, if you want to connect to a device with a specific name you have to handle the scan and manage to find the id of that peripheral.

**Reply by enzo69006** [reply]
Ok, Thank you for the quick response :)

**Reply by jrubiano01** [reply]
I'm trying to connect to a specific mobile device, I have its Bluetooh address, but when I do the scan it doesn't appear, I have to try several times? Or is it a glitch?

**Reply by enzo69006** [reply]
if you are on iOS, u can connect with the mac address
since iOS 7 or 8, ios use their own system of ' mac address '

**Comment by danielgatis** [comment]
@marcosinigaglia 

If I scan for the device and, when connecting to it, save its `peripheral.id`, can I reuse the same `peripheral.id` to connect to the device in the future without scanning again? 

If so, do you know if this strategy works on both Android and iOS?

**Comment by lucaswitch** [comment]
Yes you can.

**Comment by lucaswitch** [comment]
That's the standard way to do it. You can store on a in memory variable
like global.autoReconnectDeviceWithIds = [] and make your own business
logic with that.

**Comment by lucaswitch** [comment]
But on IOS you don't get the mac address, instead you would receive a uuid
that behaves just fine.

**Reply by danielgatis** [reply]
@lucaswitch Even if I scan and save using iPhone A and send that ID to my server, can another iPhone B connect to the same device using that same ID?

**Comment by lucaswitch** [comment]
The uuid is unique per device -> phone, so in that case won't work cause
the uuid is not meant to be saved on a server.You would have to save each
device uuid + a unique phone identifier to be able to distinguish each uuid
that your server have. A kind of hashmap of a hashmap
